tools/swift-inspect/Sources/swift-inspect/WindowsRemoteProcess.swift

```diff
@@ -233,37 +237,286 @@ internal final class WindowsRemoteProcess: RemoteProcess {
+      }
+
+      if !SetEvent(hWriteEvent) {
+        print("SetEvent failed \(GetLastError())")
+        return
       }
+    }
 
-      repeat {
-        if entry.dwFlags & DWORD(LF32_FREE) == DWORD(LF32_FREE) { continue }
-        body(swift_addr_t(entry.dwAddress), UInt64(entry.dwBlockSize))
-      } while Heap32Next(&entry)
+    let wait = WaitForSingleObject(hThread, waitTimeoutMs)
+    if wait != WAIT_OBJECT_0 {
+      print("WaitForSingleObject on LoadLibrary failed \(wait)")
+      return
+    }
+
+    var threadExitCode: DWORD = 0
+    GetExitCodeThread(hThread, &threadExitCode)
+    if threadExitCode == 0 {
+      print("LoadLibraryW failed \(threadExitCode)")
+      return
+    }
+
+    // Unload the dll and deallocate the dll path from the remote process
+    if !unloadDllAndPathRemote(dwProcessId: dwProcessId, hKernel32: hKernel32,
+                               dllPathRemote: dllPathRemote) {
+      print("Failed to unload the remote dll")
+      return
+    }
+  }
+
+  private func allocateDllPathRemote() -> UnsafeMutableRawPointer? {
+    // The path to the dll assuming it's in the same directory as swift-inspect.
+    let swiftInspectPath = ProcessInfo.processInfo.arguments[0]
+    let dllPath: String =
+        URL(fileURLWithPath: swiftInspectPath).deletingLastPathComponent().path +
+        "\\SwiftInspectClient.dll"
+    var dllPathRemote = UnsafeMutableRawPointer(bitPattern: 0)
+    var dllPathUtf16Len: UInt = 0
+    var dllPathWritten = false
+    dllPath.withCString(encodedAs: UTF16.self) {
+      // Check that the dll file exists
+      let dwAttrib = GetFileAttributesW($0)
+      if dwAttrib == INVALID_FILE_ATTRIBUTES ||
+         Int32(dwAttrib) & FILE_ATTRIBUTE_DIRECTORY != 0 {
+        print("\(dllPath) doesn't exist")
+        return
+      }
+      // Allocate memory and write dllPath in the remote process
+      dllPathUtf16Len = UInt(wcslen($0) * MemoryLayout<WCHAR>.size)
+      dllPathRemote = VirtualAllocEx(self.process,
+                                     nil,
+                                     SIZE_T(dllPathUtf16Len + 1), // +1 for `\0`
+                                     DWORD(MEM_COMMIT),
+                                     DWORD(PAGE_READWRITE))
+      if dllPathRemote == LPVOID(bitPattern: 0) {
+        print("VirtualAllocEx failed \(GetLastError())")
+        return
+      }
+      if !WriteProcessMemory(self.process, dllPathRemote, $0,
+                             SIZE_T(dllPathUtf16Len), nil) {
+        print("WriteProcessMemory failed \(GetLastError())")
+        VirtualFreeEx(self.process, dllPathRemote, 0, DWORD(MEM_RELEASE))
+        return
+      }
+      dllPathWritten = true
+    }
+    if !dllPathWritten {
+      print("Failed to write dll path in the remote process")
+      return nil
+    }
+    return dllPathRemote
+  }
+
+  private func unloadDllAndPathRemote(
+      dwProcessId: DWORD, hKernel32: HMODULE!,
+      dllPathRemote: UnsafeMutableRawPointer) -> Bool {
```

Comment:
Was this formatted by `swift-format`?

Reply:
Is there a swift-format config file that's used for the existing code? If I run it with its default config, it'd change the lines that aren't touched by this PR. Any preference?

Reply:
I don't think that I had saved it ... the default sounds like a good idea though.  I think that there may be a few places where it will break before the first parameter which is probably the only divergence the existing code has.